src/Sylius/Component/Core/Model/Order.php

```diff
@@ -298,16 +298,27 @@ public function hasPayment(BasePaymentInterface $payment)
     }
 
     /**
-     * {@inheritdoc}
+     * @param string|null $state
+     *
+     * @return BasePaymentInterface|null
      */
-    public function getLastNewPayment()
+    public function getLastPayment($state = null)
     {
         if ($this->payments->isEmpty()) {
             return null;
         }
 
-        $payment = $this->payments->filter(function (BasePaymentInterface $payment) {
-            return $payment->getState() === BasePaymentInterface::STATE_NEW;
+        if (null === $state) {
+            $lastPayment = $this->payments->last();
+            if (false === $lastPayment) {
+                return null;
+            }
+
+            return $lastPayment;
+        }
+
+        $payment = $this->payments->filter(function (BasePaymentInterface $payment) use ($state) {
+            return $payment->getState() === $state;
         })->last();
 
         return $payment !== false ? $payment : null;
```

Comment:
We've already checked that payments aren't empty, so last element will never be `false`.